role-make-version-changelog.sh

```diff
@@ -0,0 +1,198 @@
+#!/bin/bash
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2020, Red Hat, Inc.
+# SPDX-License-Identifier: MIT
+
+set -euo pipefail
+
+AUTOSKIP=${AUTOSKIP:-true}
+# Find the last tag in each role
+# Look at the git commits since that tag
+# Look at the actual changes since that tag
+# Figure out what to use for the new tag
+# Figure out what to put in the release notes for the release
+
+repo=${repo:-$( git remote get-url origin | awk -F'/' '{print $NF}' )}
+
+# To be used in conjunction with local-repo-dev-sync.sh
+# This script is called from every role
+
+view_diffs() {
+    local tag="$1"
+    local action
+    action=""
+    while [ "$action" != "q" ]; do
+        action=""
+        for hsh in $(git log --pretty=format:%h --no-merges --reverse "${tag}.."); do
+            git show --stat "$hsh"
+            read -r -p 'View full diff (y)? Start over (s)? Quit viewing diffs (q)? Next commit (n)? (default: n) ' action
+            if [ "$action" = y ]; then
+                git show "$hsh"
+            elif [ "$action" = s ] || [ "$action" = q ]; then
+                break
+            fi
+            action=""
+        done
+        if [ -z "${action}" ]; then
+            break
+        fi
+    done
+}
+
+get_main_branch() {
+    local br
+    br=$(git branch --list main)
+    if [ -n "$br" ]; then
+        echo main
+        return 0
+    fi
+    br=$(git branch --list master)
+    if [ -n "$br" ]; then
+        echo master
+        return 0
+    fi
+    echo UNKNOWN
+    return 1
+}
+
+git fetch --all
+# get the main branch
+mainbr=$(get_main_branch)
+currbr=$(git branch --show-current)
+# see if BRANCH already exists - editing an existing PR
+if [ -n "${BRANCH:-}" ]; then
+    BRANCH_EXISTS=$(git branch --list "$BRANCH")
+else # assume user wants to use the currently checked out branch
+    BRANCH="$currbr"
+fi
+if [ "$BRANCH" = "$mainbr" ]; then
+    echo ERROR: need a branch to use for the commit/PR
+    echo please set BRANCH to the branch you want to use for the PR
+    echo or git checkout the branch
+    exit 1
+fi
+if [ "$BRANCH" = "$currbr" ]; then
+    : # using current branch "$currbr"
+elif [ -n "${BRANCH_EXISTS:-}" ]; then
+    git checkout "$BRANCH"
+else
+    git checkout "$mainbr"
+    git checkout -b "$BRANCH"
+fi
+echo Using branch "$BRANCH" for the changelog commit/PR
+
+# get latest tag
+latest_tag=$(git describe --tags --abbrev=0 2> /dev/null)
+# special case for network and sshd
+allow_v=""
+case "$latest_tag" in
+v*)
+    if [ "$repo" = ansible-sshd ] || [ "$repo" = sshd ]; then
+        # sshd uses a leading v
+        allow_v=v
+    else
+        # network had a case where there where two tags for the
+        # latest - one with leading v and one without - git describe
+        # would return the one with the leading v - so had to strip
+        # it off
+        latest_tag="${latest_tag//v}"
+    fi
+    ;;
```

Comment:
If the branch is not tagged with any version, `git describe` fails as follows.
```
$ git describe --tags --abbrev=0
fatal: No names found, cannot describe anything.
$ echo $?
128
```
Could it be possible to add another question asking if an initial CHANGELOG.md file to be generated or not?
If the answer is `yes`, can we add new CHANGELOG.md having the contents something like this? (Maybe, we could ask another question what initial version to be set? And the date will be auto-filled...)
```
Changelog
=========

[#.#.#] - YYYY-MM-DD
--------------------

# Initial Release
```

Reply:
Fixed.  Latest commit works if no tags and will allow you to create the CHANGELOG.md